_posts/2022-03-21-getting-started-with-protobufs-for-typescript.md

```diff
@@ -0,0 +1,231 @@
+
+Creating the contents of your message is now as easy as:
+```js
+import { AddressBook, Person_PhoneType } from './generated/src/proto/addressBook'
+
+const myAddressBook: AddressBook = {
+  people: [
+    {
+      name: "Joe Blogs",
+      phones: [
+        {
+          phoneNumber: "0123456789",
+          phoneType: Person_PhoneType.MOBILE,
+        }
+      ]
+    },
+    {
+      name: "Jane Smith",
+      phones: [
+        {
+          phoneNumber: "0987654321",
+          phoneType: Person_PhoneType.HOME,
+        }
+      ]
+    }
+  ]
+}
+```
+This is almost identical to the original JSON object used in the [data structure section above](#define-the-data-structure-for-your-messages), how fantastic is that!  There are two differences in the above that take advantage of some awesome idiomatic Typescript and make life really easy:
+- The object type assignment `AddressBook` which allows you to use type checking for creating valid messages
+- The `phoneType` is assigned through an enum, e.g. `Person_PhoneType.MOBILE`
+
+However, if at this point you are thinking _*"Now wait up, this is not like the other tutorials I have seen. I was expecting to use a class object and function calls to create the message"*_, you can do it that way with Typescript, have a look at the [How to do this so it looks like most published tutorials](#how-to-do-this-so-it-looks-like-most-published-tutorials) section below.
+
+Great, this is the hard work done. You are most of the way there!
+
+## Serialise the Message to Protobuf
+After all of the above, this is the easy bit. Again taking instruction from the generated types file:
+
+- `const serializedMessage = AddressBook.encode(myAddressBook).finish()`
+
+## Do Something with the Serialised Message
+The how of transmitting the encoded messages is out of scope here, but you can for example transmit over gRPC or HTTP or save to file if you really want (as a lot of tutorials unhelpfully seem to do).
+
+## De-serialise the Message
+After you have transmitted or stored your message, at some point you are going to want to use its contents and you are going to have to de-serialize. Again this is really easy:
+
+- `const deSerializedMessage = AddressBook.decode(serializedMessage)`
+
+Done! You have your original `AddressBook` Typed message object again.
+
+Congrats, you survived and don’t need to spend days bashing your head against the wall.
+
+
+# Closing Thoughts
+Getting to even this point of doing the basics with Protobufs and Typescript was not a straightforward journey for me and I hope if you have read this far it might shortcut that process for you.  I am sure the docs will get better over time but Protobufs are not a beginner topic and everyone has to start somewhere.
```

Comment:
```suggestion
Getting to this point of doing the basics with Protobufs and Typescript was not a straightforward journey for me.  I hope if you have read this far it might shortcut that process for you.  I am sure the docs will get better over time but Protobufs are not a beginner topic and everyone has to start somewhere.
```

nitpick: The original reads poorly.